Data/NF/Internal.hs

```diff
@@ -51,5 +61,23 @@ newtype NF a
 type role NF nominal
 #endif
 
+-- UGH! GHC swapped the constraint order between
+-- 7.10 and 8.0. If anyone's using some beta compiler,
+-- it might throw up.
+--
+-- Very unfortunately, GHC (as of 8.0) does not allow the pattern
+-- synonym to be *matched* without an NFData context. Hopefully
+-- this will be fixed eventually.
```

Comment:
Of course it's a problem, but there's nothing we can do unless the compiler
people fix it. There's an utterly trivial fix: allow the user to supply a
type signature for the "smart constructor" distinct from its signature as a
pattern synonym (there's no need for those signatures to even be
*related*!). Maybe you can convince them.

On Jul 22, 2016 6:06 PM, "Edward Z. Yang" <notifications@github.com> wrote:

> In Data/NF/Internal.hs
> <https://github.com/ezyang/nf/pull/6#discussion_r71952966>:
>
> > @@ -51,5 +61,23 @@ newtype NF a
> >  type role NF nominal
> >  #endif
> >
> > +-- UGH! GHC swapped the constraint order between
> > +-- 7.10 and 8.0. If anyone's using some beta compiler,
> > +-- it might throw up.
> > +--
> > +-- Very unfortunately, GHC (as of 8.0) does not allow the pattern
> > +-- synonym to be *matched* without an NFData context. Hopefully
> > +-- this will be fixed eventually.
>
> Hmm, I wonder if this will cause problems. I guess it is not too
> burdensome to propagate the NFData constraint?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ezyang/nf/pull/6/files/db57fa061540b46fb54928e8131e84be376b7fe9#r71952966>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABzi_eT1yhDqwNSyPRECt8E0qvoqcuCPks5qYT72gaJpZM4JTKLc>
> .
>